version 8.9.1 (and 8.9.2-SNAPSHOT) shaded jar not working

## Description

I used 8.9.1 (or 8.9.2-SNAPSHOT version built locally from latest sources) shaded jar but it is not working. Could this be caused by the relocation added for glassfish jar? What's the solution/workaround?

## How to reproduce

Used the following test to verify:
```
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import org.apache.log4j.Level;
import org.apache.log4j.LogManager;

import com.spotify.docker.client.DefaultDockerClient;
import com.spotify.docker.client.DockerClient;
import com.spotify.docker.client.DockerClient.LogsParam;
import com.spotify.docker.client.LogStream;
import com.spotify.docker.client.exceptions.DockerCertificateException;
import com.spotify.docker.client.exceptions.DockerException;
import com.spotify.docker.client.messages.ContainerConfig;
import com.spotify.docker.client.messages.ContainerCreation;
import com.spotify.docker.client.messages.ContainerInfo;
import com.spotify.docker.client.messages.HostConfig;
import com.spotify.docker.client.messages.PortBinding;

public class DockerClientTest2
{
  private static final Logger log = LoggerFactory.getLogger(DockerClientTest2.class);
  
  public static void main(String[] args) throws DockerCertificateException, DockerException, InterruptedException
  {
  //Create a client based on DOCKER_HOST and DOCKER_CERT_PATH env vars
    LogManager.getRootLogger().setLevel(Level.INFO);
    final DockerClient docker = DefaultDockerClient.fromEnv().build();
    
   // final DockerClient docker = DefaultDockerClient.builder().
    
 // Pull an image
    docker.pull("hello-world:latest");

    // Bind container ports to host ports
    final String[] ports = {"80", "23"};
    final Map<String, List<PortBinding>> portBindings = new HashMap<>();
    for (String port : ports) {
      List<PortBinding> hostPorts = new ArrayList<>();
      hostPorts.add(PortBinding.of("0.0.0.0", port));
      portBindings.put(port, hostPorts);
    }

    // Bind container port 443 to an automatically allocated available host port.
    List<PortBinding> randomPort = new ArrayList<>();
    randomPort.add(PortBinding.randomPort("0.0.0.0"));
    portBindings.put("443", randomPort);

    final HostConfig hostConfig = HostConfig.builder().portBindings(portBindings).build();

    // Create container with exposed ports
    final ContainerConfig containerConfig = ContainerConfig.builder()
        .hostConfig(hostConfig)
        .image("hello-world:latest").exposedPorts(ports)
        .cmd("/hello")
        .build();

    final ContainerCreation creation = docker.createContainer(containerConfig);
    final String id = creation.id();

    // Inspect container
    final ContainerInfo info = docker.inspectContainer(id);

    // Start container
    docker.startContainer(id);

    LogStream logStream = docker.logs(id, LogsParam.stdout());

    log.info(logStream.readFully());

    // Kill container
    docker.killContainer(id);

    // Remove container
    docker.removeContainer(id);

    // Close the docker client
    docker.close();
    
  }
}

```

## What do you expect

Expected the test to succeed (as it does with my own docker-client library which is 8.9.0 plus my own changes which are present in 8.9.1) and produce output from hello-world:latest


## What happened instead

I see the following console output/trace:
```
Oct 03, 2017 5:17:29 PM com.spotify.docker.client.shaded.org.glassfish.hk2.utilities.reflection.Logger warning
WARNING: Cannot find a default implementation of the HK2 ServiceLocatorGenerator
Exception in thread "main" java.lang.RuntimeException: java.lang.ClassNotFoundException: Provider com.spotify.docker.client.shaded.org.glassfish.jersey.internal.RuntimeDelegateImpl could not be instantiated: java.lang.IllegalStateException: No generator was provided and there is no default generator registered
	at com.spotify.docker.client.shaded.javax.ws.rs.ext.RuntimeDelegate.findDelegate(RuntimeDelegate.java:152)
	at com.spotify.docker.client.shaded.javax.ws.rs.ext.RuntimeDelegate.getInstance(RuntimeDelegate.java:120)
	at com.spotify.docker.client.shaded.javax.ws.rs.core.UriBuilder.newInstance(UriBuilder.java:95)
	at com.spotify.docker.client.shaded.javax.ws.rs.core.UriBuilder.fromUri(UriBuilder.java:106)
	at com.spotify.docker.client.shaded.org.glassfish.jersey.client.JerseyWebTarget.<init>(JerseyWebTarget.java:81)
	at com.spotify.docker.client.shaded.org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:297)
	at com.spotify.docker.client.shaded.org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:76)
	at com.spotify.docker.client.DefaultDockerClient.resource(DefaultDockerClient.java:2599)
	at com.spotify.docker.client.DefaultDockerClient.pull(DefaultDockerClient.java:1239)
	at com.spotify.docker.client.DefaultDockerClient.pull(DefaultDockerClient.java:1224)
	at com.spotify.docker.client.DefaultDockerClient.pull(DefaultDockerClient.java:1218)
	at com.datatorrent.gateway.DockerClientTest2.main(DockerClientTest2.java:39)
Caused by: java.lang.ClassNotFoundException: Provider com.spotify.docker.client.shaded.org.glassfish.jersey.internal.RuntimeDelegateImpl could not be instantiated: java.lang.IllegalStateException: No generator was provided and there is no default generator registered
	at com.spotify.docker.client.shaded.javax.ws.rs.ext.FactoryFinder.newInstance(FactoryFinder.java:122)
	at com.spotify.docker.client.shaded.javax.ws.rs.ext.FactoryFinder.find(FactoryFinder.java:225)
	at com.spotify.docker.client.shaded.javax.ws.rs.ext.RuntimeDelegate.findDelegate(RuntimeDelegate.java:135)
	... 11 more
Caused by: java.lang.IllegalStateException: No generator was provided and there is no default generator registered
	at com.spotify.docker.client.shaded.org.glassfish.hk2.internal.ServiceLocatorFactoryImpl.internalCreate(ServiceLocatorFactoryImpl.java:308)
	at com.spotify.docker.client.shaded.org.glassfish.hk2.internal.ServiceLocatorFactoryImpl.create(ServiceLocatorFactoryImpl.java:293)
	at com.spotify.docker.client.shaded.org.glassfish.jersey.internal.inject.Injections._createLocator(Injections.java:138)
	at com.spotify.docker.client.shaded.org.glassfish.jersey.internal.inject.Injections.createLocator(Injections.java:109)
	at com.spotify.docker.client.shaded.org.glassfish.jersey.internal.RuntimeDelegateImpl.<init>(RuntimeDelegateImpl.java:63)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at com.spotify.docker.client.shaded.javax.ws.rs.ext.FactoryFinder.newInstance(FactoryFinder.java:118)
	... 13 more

```

## Software:

- `docker version`: 
```
Client:
 Version:      17.06.1-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   874a737
 Built:        Thu Aug 17 22:53:38 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.06.1-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   874a737
 Built:        Thu Aug 17 22:54:55 2017
 OS/Arch:      linux/amd64
 Experimental: true
```
- Spotify's docker-client version: 8.9.1

## Full backtrace

```text
Oct 03, 2017 5:17:29 PM com.spotify.docker.client.shaded.org.glassfish.hk2.utilities.reflection.Logger warning
WARNING: Cannot find a default implementation of the HK2 ServiceLocatorGenerator
Exception in thread "main" java.lang.RuntimeException: java.lang.ClassNotFoundException: Provider com.spotify.docker.client.shaded.org.glassfish.jersey.internal.RuntimeDelegateImpl could not be instantiated: java.lang.IllegalStateException: No generator was provided and there is no default generator registered
	at com.spotify.docker.client.shaded.javax.ws.rs.ext.RuntimeDelegate.findDelegate(RuntimeDelegate.java:152)
	at com.spotify.docker.client.shaded.javax.ws.rs.ext.RuntimeDelegate.getInstance(RuntimeDelegate.java:120)
	at com.spotify.docker.client.shaded.javax.ws.rs.core.UriBuilder.newInstance(UriBuilder.java:95)
	at com.spotify.docker.client.shaded.javax.ws.rs.core.UriBuilder.fromUri(UriBuilder.java:106)
	at com.spotify.docker.client.shaded.org.glassfish.jersey.client.JerseyWebTarget.<init>(JerseyWebTarget.java:81)
	at com.spotify.docker.client.shaded.org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:297)
	at com.spotify.docker.client.shaded.org.glassfish.jersey.client.JerseyClient.target(JerseyClient.java:76)
	at com.spotify.docker.client.DefaultDockerClient.resource(DefaultDockerClient.java:2599)
	at com.spotify.docker.client.DefaultDockerClient.pull(DefaultDockerClient.java:1239)
	at com.spotify.docker.client.DefaultDockerClient.pull(DefaultDockerClient.java:1224)
	at com.spotify.docker.client.DefaultDockerClient.pull(DefaultDockerClient.java:1218)
	at com.datatorrent.gateway.DockerClientTest2.main(DockerClientTest2.java:39)
Caused by: java.lang.ClassNotFoundException: Provider com.spotify.docker.client.shaded.org.glassfish.jersey.internal.RuntimeDelegateImpl could not be instantiated: java.lang.IllegalStateException: No generator was provided and there is no default generator registered
	at com.spotify.docker.client.shaded.javax.ws.rs.ext.FactoryFinder.newInstance(FactoryFinder.java:122)
	at com.spotify.docker.client.shaded.javax.ws.rs.ext.FactoryFinder.find(FactoryFinder.java:225)
	at com.spotify.docker.client.shaded.javax.ws.rs.ext.RuntimeDelegate.findDelegate(RuntimeDelegate.java:135)
	... 11 more
Caused by: java.lang.IllegalStateException: No generator was provided and there is no default generator registered
	at com.spotify.docker.client.shaded.org.glassfish.hk2.internal.ServiceLocatorFactoryImpl.internalCreate(ServiceLocatorFactoryImpl.java:308)
	at com.spotify.docker.client.shaded.org.glassfish.hk2.internal.ServiceLocatorFactoryImpl.create(ServiceLocatorFactoryImpl.java:293)
	at com.spotify.docker.client.shaded.org.glassfish.jersey.internal.inject.Injections._createLocator(Injections.java:138)
	at com.spotify.docker.client.shaded.org.glassfish.jersey.internal.inject.Injections.createLocator(Injections.java:109)
	at com.spotify.docker.client.shaded.org.glassfish.jersey.internal.RuntimeDelegateImpl.<init>(RuntimeDelegateImpl.java:63)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at java.lang.Class.newInstance(Class.java:442)
	at com.spotify.docker.client.shaded.javax.ws.rs.ext.FactoryFinder.newInstance(FactoryFinder.java:118)
	... 13 more

```

--- a/pom.xml
+++ b/pom.xml
@@ -309,7 +309,7 @@
       <plugin>
         <groupId>org.apache.maven.plugins</groupId>
         <artifactId>maven-shade-plugin</artifactId>
-        <version>2.4.1</version>
+        <version>3.1.0</version>
         <configuration>
           <createSourcesJar>true</createSourcesJar>
           <artifactSet>
@@ -361,6 +361,9 @@
             </relocation>
           </relocations>
           <shadedArtifactAttached>true</shadedArtifactAttached>
+          <transformers>
+            <transformer implementation="org.apache.maven.plugins.shade.resource.ServicesResourceTransformer"/>
+          </transformers>
         </configuration>
         <executions>
           <execution>

